admin/controllers/pupil-pin.js

```diff
@@ -152,6 +152,51 @@ const getViewAndPrintPins = async (req, res, next) => {
   })
 }
 
+const getViewAndCustomPrintPins = async (req, res, next) => {
+  const pinEnv = (req.params && req.params.pinEnv === 'live') ? 'live' : 'familiarisation'
+  res.locals.pinEnv = pinEnv
+  res.locals.pageTitle = `View and custom print PINs`
+  req.breadcrumbs(
+    `PINs for ${pinEnv} check`,
+    `/pupil-pin/generate-${pinEnv}-pins-overview`)
+  req.breadcrumbs(res.locals.pageTitle)
+
+  const helplineNumber = config.Data.helplineNumber
+  let pupils
+  let groupsForPupils
+  let groups
+  let school
+  let error
+  const date = dateService.formatDayAndDate(new Date())
+  try {
+    pupils = await pinService.getPupilsWithActivePins(req.user.School, pinEnv)
+    groupsForPupils = await groupService.getGroupsAsArray(req.user.schoolId)
+    school = await pinService.getActiveSchool(req.user.School)
+    error = await checkWindowSanityCheckService.check()
+    if (pupils.length > 0) {
```

Comment:
Could avoid the nested if here with a guard clause.
Also I would think this logic could be part of a service?